### PR TITLE
Update GioFuture to not require fragile crate usage by all users anymore

### PR DESCRIPTION
**Pull request by sdroege**
See gtk-rs/gio#257

Depends on https://github.com/gtk-rs/gir/pull/875

### CODE REVIEW

**Comment by sdroege** [comment]
@GuillaumeGomez @EPashkin Opinions?

**Comment by sdroege** [comment]
Also makes all generated futures code half the size or something :)

**Comment by EPashkin** [comment]
Looks good, but:
```
error[E0463]: can't find crate for `fragile`
  --> src\lib.rs:20:1
   |
20 | extern crate fragile;
   | ^^^^^^^^^^^^^^^^^^^^^ can't find crate
error: aborting due to previous error

```

**Comment by EPashkin** [comment]
and fragile still used in
```
pollable_input_stream.rs
pollable_output_stream.rs
socket.rs
```

**Comment by sdroege** [comment]
Thanks! This depends on https://github.com/gtk-rs/glib/pull/542 now too. There was some unneeded fragile usage left indeed, and I could also clean up some more code related to that now.

File: src/pollable_input_stream.rs [hunk 1]
```diff
@@ -166,16 +162,15 @@ impl<O: IsA<PollableInputStream>> PollableInputStreamExtManual for O {
...
-            obj.get()
-                .create_source(cancellable.as_ref(), None, priority, move |_| {
-                    if send.unbounded_send(()).is_err() {
-                        glib::Continue(false)
-                    } else {
-                        glib::Continue(true)
-                    }
-                })
+            obj.create_source(cancellable.as_ref(), None, priority, move |_| {
+                if send.unbounded_send(()).is_err() {
```
Comment (GuillaumeGomez) [inline src/pollable_input_stream.rs]: Why not `glib::Continue(send.unbounded_send(()).is_ok())`?

File: src/pollable_output_stream.rs [hunk 2]
```diff
@@ -138,17 +134,16 @@ impl<O: IsA<PollableOutputStream>> PollableOutputStreamExtManual for O {
...
-            obj.get()
-                .create_source(cancellable.as_ref(), None, priority, move |_| {
-                    if send.as_ref().unwrap().unbounded_send(()).is_err() {
-                        glib::Continue(false)
-                    } else {
-                        glib::Continue(true)
-                    }
-                })
+            obj.create_source(cancellable.as_ref(), None, priority, move |_| {
+                if send.as_ref().unwrap().unbounded_send(()).is_err() {
```
Comment (GuillaumeGomez) [inline src/pollable_output_stream.rs]: Same.

File: src/pollable_input_stream.rs (hunk 1 above)
Comment (sdroege) [reply src/pollable_input_stream.rs]: Harder to read :)

File: src/pollable_output_stream.rs (hunk 2 above)
Comment (sdroege) [reply src/pollable_output_stream.rs]: Same ;)

**Pull request merged**